src/main/java/frc/robot/RobotContainer.java

```diff
@@ -83,7 +84,14 @@ private void configureButtonBindings() {
     // runs the shoot one ball command while A is held
     new JoystickButton(m_operatorController, Button.kA.value).whenHeld(new ShootOneBallCommand(m_shooterSubsystem));
 
-    new JoystickButton(m_driveController, Button.kStart.value).whenPressed(new ResetGyroCommand(m_swerveSubsystem));
+    // resets the gyro when the button is pressed
+    new JoystickButton(m_driveController, Button.kStart.value)
+        .whenPressed(new InstantCommand(m_swerveSubsystem::resetGyro, m_swerveSubsystem));
+    // Sets brake and coast mode
+    new JoystickButton(m_driveController, Button.kBumperLeft.value)
+        .whenPressed(() -> m_swerveSubsystem.setDriveIdleMode(IdleMode.kCoast));
+    new JoystickButton(m_driveController, Button.kBumperLeft.value)
+        .whenReleased(() -> m_swerveSubsystem.setDriveIdleMode(IdleMode.kBrake));
   }
```

Comment:
minor note, but we will have to properly clean up button bindings later. the left bumper is currently used for shooteroncommand in main.